File: log.js
var log = (function() {
	var log = {
		// log level
		level: 0,
		defaultLevel: 0,
		groups: {},
		log: function(content, level) {
			// set default level if none supplied
			if(arguments.length === 1) {
				level = this.defaultLevel;
			}
			// if level is a number,
			if(typeof level === 'number') {
				// print contents if current level is lower
				if(level >= this.level) {
					console.log(content);
				}
			} else {
				// if level is not a number, print contents
				// if debug group is enabled
				if(groups[level]) {
					console.log(content);
				}
			}
		}, 
		enable: function(level) {
			groups[level] = true;
		}
		disable: function(level) {
			groups[level] = false;
		}
	};
}());

File: dart/Log.dart
/** [Log] is a class for logging
*  based on log level or groups
**/
#library("Log");

class Log {
  // the current logging level
  int level;
  // the set of currently active groups
  Set<String> groups;

  Log([int level = 0, Collection<String> groups = null]) {
    this.level = level;
    if(groups != null) {
      this.groups = new Set.from(groups);
    } else {
      this.groups = new Set();
    }
  }

  bool Log(content, [int level, String group]) {
    if(level != null) {
      if(level <= this.level) {
        print(content);
        return true;
      }
    } else {
      if(group != null) {
        if(groups.contains(group)) {
          print(content);
          return true;
        }
      } else {
        print(content);
        return true;
      }
    }
    return false;
  }
}